### PR TITLE
Paper - Courtney Massey

### PR DESCRIPTION
**Pull request by courtneycmassey**

### CODE REVIEW

File: swap_meet/clothing.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from .item import Item
+
+class Clothing(Item):
```
Comment (CheezItMan) [inline swap_meet/clothing.py:3]: 👍 Very clean child classes!

File: swap_meet/item.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+class Item:
+    
+    def __init__(self, category="", condition=None, age=None):
+        if category == "":
+            self.category = ""
```
Comment (CheezItMan) [inline swap_meet/item.py:5]: I don't think you need this if statement.

File: swap_meet/item.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        else:
+            self.category = category
+        if condition == None:
+            self.condition = 0
+        else:
+            self.condition = condition 
+        if age == None:
+            self.age = None
+        else:
+            self.age = age
```
Comment (CheezItMan) [inline swap_meet/item.py:15]: I also don't think you need this if-else

File: swap_meet/vendor.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+class Vendor:
+
+    def __init__(self, inventory=None):
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory 
+
+    def add(self, new_item):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:11]: 👍

File: swap_meet/vendor.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        self.inventory.append(new_item)
+        return new_item
+    
+    def remove(self, item):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:15]: 👍

File: swap_meet/vendor.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+    
+    def get_by_category(self, category):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:22]: 👍

File: swap_meet/vendor.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        items_in_category = []
+        for item in self.inventory:
+            if item.category == category:
+                items_in_category.append(item)
+        return items_in_category
+    
+    def swap_items(self, other_vendor, my_item, their_item):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:29]: 👍

File: swap_meet/vendor.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.remove(my_item)
+            other_vendor.add(my_item)
+            other_vendor.remove(their_item)
+            self.add(their_item)
+            return True
+        else:
+            return False
```
Comment (CheezItMan) [inline swap_meet/vendor.py:37]: Minor comment, the else isn't needed here.

```suggestion
        return False
```

File: swap_meet/vendor.py [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+    
+    def swap_first_item(self, other_vendor):        
+        if len(self.inventory) == 0 or len(other_vendor.inventory) == 0:
```
Comment (CheezItMan) [inline swap_meet/vendor.py:40]: You can also do this this way because `[]` is falsey.

```suggestion
        if not self.inventory or not other_vendor.inventory:
```

File: swap_meet/vendor.py [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+            return False
+        else:
+            self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
+            return True
+
+    def get_best_by_category(self, category):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:46]: 👍

File: swap_meet/vendor.py [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+        items_in_category = self.get_by_category(category)
+        if len(items_in_category) == 0:
+            return None
+        else:
```
Comment (CheezItMan) [inline swap_meet/vendor.py:50]: Another unneeded `else`

File: swap_meet/vendor.py [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+            best_condition = 0
+            for item in items_in_category:
+                if item.condition >= best_condition:
+                    best_condition = item.condition
+                    best_item = item
+            return best_item
+
+    def swap_best_by_category(self, other, my_priority, their_priority):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:58]: 👍

File: swap_meet/vendor.py [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+        vendor_priority_item = other.get_best_by_category(my_priority)
+        other_priority_item = self.get_best_by_category(their_priority)
+        if vendor_priority_item == None or other_priority_item == None:
+            return False
+        else:
+            self.swap_items(other, other_priority_item, vendor_priority_item)
+            return True
```
Comment (CheezItMan) [inline swap_meet/vendor.py:65]: You can simplify this a bit.  Just a suggestion.

```suggestion
        if not vendor_priority_item or not other_priority_item:
            return False

        self.swap_items(other, other_priority_item, vendor_priority_item)
        return True
```

File: swap_meet/vendor.py [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+
+
+    #optional enhancements
+    def get_newest(self):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:69]: 👍

File: swap_meet/vendor.py [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+        age_of_newest = 100000
+        for item in self.inventory:
+            if item.age == None:
+                continue
+            elif item.age < age_of_newest:
+                newest_item = item
+                age_of_newest = item.age
+        return newest_item
+
+    def swap_by_newest(self, other):
```
Comment (CheezItMan) [inline swap_meet/vendor.py:80]: 👍 , similar suggestions with regard to the if-else statement.

File: tests/test_wave_07.py [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+def test_items_have_age_as_float():
+    items = [
+        Clothing(age=15.0),
+        Decor(age=15.0),
+        Electronics(age=15.0)
+    ]
+    for item in items:
+        assert item.age == pytest.approx(15.0)
+
+def test_get_newest_returns_youngest():
```
Comment (CheezItMan) [inline tests/test_wave_07.py:17]: 👍

File: tests/test_wave_07.py [hunk 17]
```diff
@@ -0,0 +1,94 @@
...
+        inventory=[item_a, item_b, item_c]
+    )
+
+    newest_item = tai.get_newest()
+
+    assert newest_item.age == 6
+    assert newest_item == item_b
+
+
+def test_swap_by_newest():
```
Comment (CheezItMan) [inline tests/test_wave_07.py:47]: 👍

File: tests/test_wave_07.py [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+
+    assert result is True
+    assert len(tai.inventory) is 3
+    assert len(jesse.inventory) is 3
+    assert item_c not in tai.inventory
+    assert item_e in tai.inventory
+    assert item_e not in jesse.inventory
+    assert item_c in jesse.inventory
+
+def test_swap_by_newest_no_age_is_false():
```
Comment (CheezItMan) [inline tests/test_wave_07.py:74]: 👍 Nice edge-case test

**Review by CheezItMan — COMMENTED** [review]
Very well done Courtney, you hit the learning goals here.  I had a few suggestions, mostly on coding style, but this is very well done.  Nice work.
